qutebrowser/browser/history.py

```diff
@@ -452,6 +452,63 @@ def history_clear(force=False):
                               title="Clear all browsing history?")
 
 
+def _normalize_url(url: str) -> QUrl:
+    if not url:
+        return QUrl()
+    if '://' not in url:
+        url = f'https://{url}'
+    return QUrl(url)
+
+
+def _find_matching_history_urls(qurl: QUrl) -> list[str]:
+    if not qurl.isValid():
+        return []
+
+    target_host = qurl.host().lower()
+    matching_urls = []
+
+    for entry in web_history:
+        entry_qurl = QUrl(entry.url)
+        if entry_qurl.host().lower() == target_host:
```

Comment:
You say *e.g., "example.com", "https://example.com/page"* below, yet this simply gets the host out of the URL and then still deletes everything matching that host...

Reply:
yup, my initial run through i was trying to match page and domain, but then was thinking that domain made more sense. so great callout as well.